runtime/CSharp/runtime/CSharp/Antlr4.Runtime/Atn/SingletonPredictionContext.cs

```diff
@@ -79,7 +79,7 @@ public override bool Equals(object o)
                 return false;
             }
 			Antlr4.Runtime.Atn.SingletonPredictionContext other = (Antlr4.Runtime.Atn.SingletonPredictionContext)o;
-            return returnState == other.returnState && parent.Equals(other.parent);
+            return returnState == other.returnState && (parent != null && parent.Equals(other.parent));
```

Comment:
How about when parent is null?

Reply:
That is the change i want to add, when parent == null i want to return false. before we did a compare on null and got a nullrefexception. This is the same check that is done in java code SingletonPredictionContext.java.

Reply:
Then maybe the Java code also needs to be fixed. Please get this right - maybe add a reusable utility to compare objects.

Reply:
It looks like what the Java target does and I'm terrified of changing what that target does so maybe we should allow this within the C# to match the functionality at least?